haystack/utils/openai_utils.py

```diff
@@ -74,14 +74,24 @@ def _openai_text_completion_tokenization_details(model_name: str):
 
     :param model_name: Name of the OpenAI model.
     """
-    tokenizer_name = "gpt2"
-    if "davinci" in model_name:
-        max_tokens_limit = 4000
-        if USE_TIKTOKEN:
-            tokenizer_name = MODEL_TO_ENCODING.get(model_name, "p50k_base")
+    # Adding the new tokenizer for ChatGPT model and adjusting the `max_tokens_limit`
+    ## Ref: https://github.com/openai/tiktoken/blob/d0cc25c8a670941e38062cd67eef97c7f6bb74d3/tiktoken/model.py#L14
```

Comment:
I appreciate the comments for explaining your changes, but I think these would be best moved to your PR description in GH and taken out of the source code. For example, the tokenizer for the ChatGPT model will eventually no longer be new 😉

Reply:
I think you are correct! I need to write general comment for this.